src/main/java/com/hazelcast/eureka/one/EurekaOneDiscoveryStrategy.java

```diff
@@ -225,12 +224,17 @@ private EurekaInstanceConfig buildInstanceConfig(DiscoveryNode localNode) {
                 String key = String.format("%s.datacenter", this.namespace);
                 value = props.getProperty(key, "");
             } else {
-                value = String.valueOf(getProperties().get("datacenter"));
+                value = String.valueOf(getProperties().get(DATACENTER.key()));
             }
             if ("cloud".equals(value.trim().toLowerCase())) {
                 return new DelegatingInstanceConfig(new CloudInstanceConfig(this.namespace), localNode);
             }
-            return new DelegatingInstanceConfig(new MyDataCenterInstanceConfig(this.namespace), localNode);
+            if (this.useClasspathEurekaClientProps) {
+                return new DelegatingInstanceConfig(new MyDataCenterInstanceConfig(this.namespace), localNode);
+            }
+            else {
+                return new DelegatingInstanceConfig(new MyDataCenterInstanceConfig(this.namespace), localNode, getAppname());
```

Comment:
Do you think it would be possible (in a simple manner) to add a unit test to cover this line you added?

Reply:
Not so easily, at the moment mocks are used throughout tests, thus preventing real testing of instance-config classes, not only this one. I'll add an issue.